app/importers/file_importers/attendance_importer.rb

```diff
@@ -43,30 +40,39 @@ def import
     log("@skipped_old_rows_count: #{@skipped_old_rows_count}")
     log("@skipped_other_rows_count: #{@skipped_other_rows_count}")
     log('')
-    log("@unchanged_rows_count: #{@unchanged_rows_count}")
-    log("@updated_rows_count: #{@updated_rows_count}")
-    log("@created_rows_count: #{@created_rows_count}")
-    log("@invalid_rows_count: #{@invalid_rows_count}")
-  end
 
-  def remote_file_name
-    LoadDistrictConfig.new.remote_filenames.fetch('FILENAME_FOR_ATTENDANCE_IMPORT', nil)
+    log("Removing records within scope that didn't have CSV rows...")
+    @tardy_record_syncer.delete_unmarked_records!(records_within_scope(Tardy))
+    @absence_record_syncer.delete_unmarked_records!(records_within_scope(Absence))
   end
 
-  def client
-    SftpClient.for_x2
+  private
+  def download_csv
+    client = SftpClient.for_x2
+    remote_file_name = LoadDistrictConfig.new.remote_filenames.fetch('FILENAME_FOR_ATTENDANCE_IMPORT', nil)
+    data_transformer = StreamingCsvTransformer.new(@log)
+    CsvDownloader.new(
+      log: @log,
+      remote_file_name: remote_file_name,
+      client: client,
+      transformer: data_transformer
+    ).get_data
   end
 
-  def data_transformer
-    StreamingCsvTransformer.new(@log)
+  # What existing Insights records should be updated or deleted from running this import?
+  def records_within_scope(record_class)
+    record_class
+      .joins(:student => :school)
+      .where(:schools => {:local_id => @school_local_ids})
+      .where('occurred_at >= ?', @old_threshold)
   end
 
   def school_filter
-    SchoolFilter.new(@school_scope)
+    SchoolFilter.new(@school_local_ids)
   end
 
   def is_old?(row)
```

Comment:
I wrote a "note to self" here as I reviewed saying that if the event date falls exactly on the `@old_threshold`, the record is considered not old. That made it easy for me to look at the `records_within_scope` query and verify that records falling on the threshold are correctly handled as within scope.